packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -146,32 +263,171 @@ class TimeInput extends React.Component {
 
     // If the change made was not a deletion of a digit,
     // then prepend '0' if the input value is a single digit value between 3 and 9 for hour or 6 and 9 for minute.
+
     if (inputValue.length >= stateValue.length) {
-      const digitsToPrependZero = type === inputType.HOUR ? ['3', '4', '5', '6', '7', '8', '9'] : ['6', '7', '8', '9'];
+      const digitsToPrependZero = ['3', '4', '5', '6', '7', '8', '9'];
+
+      if (this.props.variant === FORMAT_12_HOUR) {
+        digitsToPrependZero.push('2');
+      }
+
       if (digitsToPrependZero.indexOf(inputValue) > -1) {
         inputValue = '0'.concat(inputValue);
       }
     }
 
+    if (inputValue === '00') {
+      inputValue = '12';
+    }
+
     // // Move focus to the minute input if the hour input has a valid and complete entry.
-    if (type === inputType.HOUR && inputValue.length === 2) {
+    if (inputValue.length === 2) {
       this.minuteInput.textInput.focus();
     }
 
-    this.handleValueChange(event, type, inputValue);
+    this.handleValueChange(event, inputType.HOUR, inputValue, this.state.meridiem);
   }
 
-  handleHourInputKeyDown(event) {
-    this.handleInputKeyDown(event, inputType.HOUR);
+  handleMinuteChange(event) {
+    // Ignore the input and don't update the state if the entered value is a non numeric character.
+    if (event.target.value.length > 0) {
```

Comment:
What is this included for? I notice the hour doesn't check this?

Reply:
It's in both. It was code left over from last time, so it didn't show up in the check. https://github.com/cerner/terra-core/blob/master/packages/terra-time-input/src/TimeInput.jsx#L129

I'm looking at moving it to TimeUtil, but I can't think of a good method name for this code. It's essentially exiting the function if the value is present, but not numeric. I'll think on it more , but given that it's just a few line in 2 locations, I don't think it's too bad to have a little duplication.

Reply:
possible name could be `validateNumericInput`